libyul/AsmParser.cpp

```diff
@@ -158,43 +158,49 @@ Statement Parser::parseStatement()
 		}
 		while (currentToken() == Token::Comma);
 
-		expectToken(Token::Colon);
-		expectToken(Token::Assign);
+		expectToken(Token::AssemblyAssign);
+
+		assignment.value.reset(new Expression(parseExpression()));
+		assignment.location.end = locationOf(*assignment.value).end;
+		return Statement{std::move(assignment)};
+	}
+	case Token::AssemblyAssign:
```

Comment:
Actually it should be possible to combine this case with the one above - since they are basically

 - ( Identifier "," )* Identifier ":=" Expression
 - Identifier ":=" Expression

Also interesting: In the second case we check `m_builtins`, but we do not in the first one. Would be interesting how the compiler currently behaves for `x, return := f()`